planet/src/main/java/com/google/planet/servlets/ItineraryServlet.java

```diff
@@ -0,0 +1,72 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.planet.servlets;
+import com.google.gson.Gson;
+
+import com.google.planet.data.Event;
+import com.google.planet.data.ItineraryGenerator;
+import com.google.planet.data.ItineraryItem;
+import com.google.planet.data.TimeRange;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.planet.data.Event;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.List;
+import java.lang.*; 
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that handles the game two truths and one lie */
+
+@WebServlet("/generate-itinerary")
+public class ItineraryServlet extends HttpServlet {
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Query query = new Query("Event");
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+
+        // Get list of events from Datastore
+        List<Event> events = new ArrayList<>();
+        for (Entity entity : results.asIterable()) {
+            long id = entity.getKey().getId();
+            String name = (String) entity.getProperty("name");
+            String address = (String) entity.getProperty("address");
+            double duration = (double) entity.getProperty("duration");
+            int openingTime = Math.toIntExact((long)entity.getProperty("openingTime"));
+            int closingTime = Math.toIntExact((long)entity.getProperty("closingTime"));
+            String listName = (String) entity.getProperty("listName");
+            String userId = (String) entity.getProperty("userId");
+            Event event = new Event(id, name, address, duration, 
+                TimeRange.fromStartEnd(openingTime, closingTime), listName, userId);
+            events.add(event);
+        }
+
```

Comment:
Big part of this doPost is exactly the same as the doGet function of the EventServlet, Is it possible to avoid code duplication in this case?

Reply:
Hmmm I searched up sending responses from one servlet to another. Although it's possible, since eventservlet is run everytime the page loads but itinerary servlet is only run when the user clicks generate itinerary button, I'm not sure if sending requests between servlets are the best option

Reply:
Is it possible to get the current events' data from the client side since they already on the DOM?